scripts/upgrade_rooms.py

```diff
@@ -282,21 +309,76 @@ def main() -> None:
     public_rooms = get_public_rooms(admin_access_token, server_url, limit=args.limit)
     if args.dry_run:
         logger.info("[DRY-RUN] Listing up to %d public rooms from %s", args.limit, server_url)
+        print("room_id;name;total_members;version;error")
         for room in public_rooms:
             name = room.get("name", "")
             room_id = room.get("room_id", "")
             num_joined = room.get("num_joined_members", 0)
-            print(f"{name};{room_id};{num_joined}")
+            error = ""
+            if homeserver not in room_id:
+                continue
+            try:
+                room_version = get_room_version(admin_access_token, server_url, room_id)
+            except SynapseServerError:
+                error = "Failed"
+            print(f"{name};{room_id};{num_joined};{room_version};{error}")
         return
 
     if args.room_id:
-        logger.info("room-id is set, upgrading %s", args.room_id)
+        logger.info("%s - upgrading using room_id", args.room_id)
+        current_room_version = get_room_version(admin_access_token, server_url, args.room_id)
+        if current_room_version == args.version:
```

Comment:
Minor: `version = args.version` is defined in line 302.
```suggestion
        if current_room_version == version:
```